webapp/tests/karma/ts/services/rules-engine.service.spec.ts

```diff
@@ -0,0 +1,619 @@
+    fetchTasksForRecursive = sinon.stub();
+    fetchTasksFor.events = {};
+    fetchTasksForRecursive.callsFake((event, fn) => {
+      fetchTasksFor.events[event] = fetchTasksFor.events[event] || [];
+      fetchTasksFor.events[event].push(fn);
+      const promise = fetchTasksResult();
+      promise.on = fetchTasksForRecursive;
+      return promise;
+    });
+    fetchTasksFor.returns({ on: fetchTasksForRecursive });
+
+    fetchTargetsResult = Promise.resolve;
+    fetchTargets = sinon.stub();
+    fetchTargets.events = {};
+    fetchTargetsRecursive = sinon.stub();
+    fetchTargetsRecursive.callsFake((event, fn) => {
+      fetchTargets.events[event] = fetchTargets.events[event] || [];
+      fetchTargets.events[event].push(fn);
+      const promise = fetchTargetsResult();
+      promise.on = fetchTargetsRecursive;
+      return promise;
+    });
+    fetchTargets.returns({ on: fetchTargetsRecursive });
+
+    rulesEngineCoreStubs = {
+      initialize: sinon.stub().resolves(true),
+      isEnabled: sinon.stub().returns(true),
+      fetchTasksFor: fetchTasksFor,
+      fetchTargets: fetchTargets,
+      updateEmissionsFor: sinon.stub().resolves(true),
+      rulesConfigChange: sinon.stub().returns(true),
+      getDirtyContacts: sinon.stub().returns([]),
+    };
+    const rulesEngineCoreFactory= { get: () => rulesEngineCoreStubs };
+
+    TestBed.configureTestingModule({
+      imports: [
+        TranslateModule.forRoot({ loader: { provide: TranslateLoader, useClass: TranslateFakeLoader } }),
+      ],
+      providers: [
+        ParseProvider,
+        ContactTypesService,
+        { provide: AuthService, useValue: authService },
+        { provide: SessionService, useValue: sessionService },
+        { provide: SettingsService, useValue: settingsService },
+        { provide: TelemetryService, useValue: telemetryService },
+        { provide: UHCSettingsService, useValue: uhcSettingsService },
+        { provide: UserContactService, useValue: userContactService },
+        { provide: UserSettingsService, useValue: userSettingsService },
+        { provide: ChangesService, useValue: changesService },
+        { provide: TranslateFromService, useValue: translateFromService },
+        { provide: RulesEngineCoreFactoryService, useValue: rulesEngineCoreFactory },
+        { provide: PipesService, useValue: pipesService }
+      ]
+    });
+
+    authService = TestBed.inject(AuthService);
+    sessionService = TestBed.inject(SessionService);
+    settingsService = TestBed.inject(SettingsService);
+    telemetryService = TestBed.inject(TelemetryService);
+    uhcSettingsService = TestBed.inject(UHCSettingsService);
+    userContactService = TestBed.inject(UserContactService);
+    userSettingsService = TestBed.inject(UserSettingsService);
+    changesService = TestBed.inject(ChangesService);
+    contactTypesService = TestBed.inject(ContactTypesService);
+    translateFromService = TestBed.inject(TranslateFromService);
```

Comment:
I don't think you need to assign all these. Just the service you're testing.

Reply:
Done